### PR TITLE
Sp1 us2

### PR DESCRIPTION
**Pull request by dan-rds**
From US2:
- Unified theme in context in relation to colors and fonts (2) (Daniel)
- Arrow painter UI update based on Product Owner’s mockup (2) (Daniel)
- List angle UI (5)
- Raw data panel of boat speed, wind speed, list angle (3)
From US1:
- Converted UI to stream builders

Along the way I also made a list angle service, provider and model etc.

Also changed all the streamcontrollers in the bloc to behavior subjects.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -68,4 +68,5 @@ fabric.properties
 gradle
 gradlew
 gradlew.bat
-.gradle
\ No newline at end of file
+.gradlei
```
Comment (willnwalker) [inline .gitignore]: Was this intentional? It's fine if so, wasn't aware Gradle had renamed their settings directory

File: binnacle/lib/screens/test_screen.dart [hunk 2]
```diff
@@ -29,53 +30,54 @@ Widget swapCompassButton(Bloc bloc) {
...
-      } else if (snapshot.hasError) {
-        return Text('**POSISTION** Hmmm... something went wrong');
-      } else {
-        return Text('**POSISTION** No data yet');
-      }
-    }
-  );
+          ]);
+        } else if (snapshot.hasError) {
+          return Text('**POSISTION** Hmmm... something went wrong');
```
Comment (willnwalker) [inline binnacle/lib/screens/test_screen.dart]: Typo here, should be position

File: binnacle/lib/services/weather_service.dart [hunk 3]
```diff
@@ -16,29 +16,43 @@ class WeatherService extends IWindService {
...
     if (response.statusCode == 200) {
+      // TODO remove this debugging code
       // If the call to the server was successful, parse the JSON
       var temp = WeatherModel.fromJson(json.decode(response.body));
       WindModel wind = WindModel(temp.wind.speed, temp.wind.deg);
       print("Temp:" + temp.toString());
-      print(temp.wind.speed.toString() + "   " +  temp.wind.deg.toString());
+      print(temp.wind.speed.toString() + "   " + temp.wind.deg.toString());
       _windStream.sink.add(wind);
     } else {
```
Comment (willnwalker) [inline binnacle/lib/services/weather_service.dart]: Did you forget to add an error? We can implement it later, but should add a TODO

Comment (dan-rds) [reply binnacle/lib/services/weather_service.dart]: Added a TODO

File: binnacle/lib/screens/test_screen.dart (hunk 2 above)
Comment (dan-rds) [reply binnacle/lib/screens/test_screen.dart]: Good catch, fixed it

**Pull request merged**